yb-voyager/src/query/queryissue/detectors_ddl.go

```diff
@@ -603,6 +603,25 @@ func (v *ViewIssueDetector) DetectIssues(obj queryparser.DDLObject) ([]QueryIssu
 	return issues, nil
 }
 
+// ================FUNCTION ISSUE DETECTOR ==================
+
+type FunctionIssueDetector struct{}
+
+func (v *FunctionIssueDetector) DetectIssues(obj queryparser.DDLObject) ([]QueryIssue, error) {
+	function, ok := obj.(*queryparser.Function)
+	if !ok {
+		return nil, fmt.Errorf("invalid object type: expected View")
+	}
+	var issues []QueryIssue
+
+	if function.HasSqlBody {
```

Comment:
- Can you use just `BEGIN` to start a function block? 
- If so, are both `BEGIN` and `BEGIN ATOMIC` unsupported? 
- ~~sql_body could also just be a "return expression" from docs. Would that also be unsupported? https://www.postgresql.org/docs/current/sql-createfunction.html#:~:text=a%20new%20session.-,sql_body,is%20more%20compatible%20with%20the%20SQL%20standard%20and%20other%20SQL%20implementations.,-Overloading~~ I see in your test that that is also unsupported.

Reply:
> Can you use just BEGIN to start a function block?

yes BEGIN can also start a function block

> If so, are both BEGIN and BEGIN ATOMIC unsupported?

No only `BEGIN ATOMIC` is unsupported as its PG 15 feature

Reply:
Right, but will `function.HasSqlBody` be true even if the function has just `BEGIN` ? That would be a problem, right?